src/Tui.hs

```diff
@@ -98,12 +102,21 @@ msgSchema = MessageSchema { msgSchemaName = MN "SearchRequest"
                             }]
                           }
 
-processFTV :: String -> String -> FieldValue
+data FieldChoice = FReq | FOpt | FRep | Wrong deriving (Show, Read)
+
+toFC :: String -> FieldChoice
+toFC s = case s of
+    "query"    -> FReq
+    "maybeNum" -> FOpt
+    "numList"  -> FRep
+    _          -> Wrong
+
+processFTV :: FieldChoice -> String -> Either String FieldValue
 processFTV field value = case field of
-    "query"    -> FText (Req $ read value)
-    "maybeNum" -> FInt  (Opt $ read value)
-    "numList"  -> FInt  (Rep $ read value)
-    _   -> FText Error
+    FReq  -> readEither $ show (FText (Req $ pack value))
+    FOpt  -> readEither $ show (FInt  (Opt $ read value))
```

Comment:
This looks weird. I expect this function to behave something like:

```haskell
FOpt -> FInt . Opt <$> readEither value
```

And similar for other cases